airflow/docker.go

```diff
@@ -106,6 +136,25 @@ func generateConfig(projectName, airflowHome string, envFile string) (string, er
 	return buff.String(), nil
 }
 
+func generateSettings() (SettingsConfig, error) {
+	var settings SettingsConfig
+	file := filepath.Join(config.WorkingPath, "settings.yaml")
+
+	buff, err := ioutil.ReadFile(file)
+
+	if err != nil {
+		return settings, errors.Wrap(err, "failed to read settings file")
+	}
+
+	err = yaml.Unmarshal(buff, &settings)
+
+	if err != nil {
+		return settings, errors.Wrap(err, "failed to generate settings")
+	}
+
+	return settings, nil
+}
+
 func checkServiceState(serviceState, expectedState string) bool {
```

Comment:
@benjamingregory this function could probably be broken out into some more modular chunks to help avoid the deep indentation.